### PR TITLE
Argument system for commands

### PR DESCRIPTION
**Pull request by aepic999**
I added an argument system for commands.
In PluginHandler.cs, I allow the Run() function to accept an extra argument, which will be passed into all functions. (NOTE: ALL plugins will break because of this unless patched to accept this string argument. An easy way to fix this is to add `string args = null` inside of the parenthesis of all of your functions. Even init is affected by this! I could have fixed the problem for init by adding a simple if statement, but everything else would still be affected.)
Shell.cs was also patched to pass arguments to the function.

### CODE REVIEW

**Comment by aepic999** [comment]
BTW the 4 extra commits were included for some reason. Those shouldn't really matter tho

File: jvyterm/PluginHandler.cs [hunk 1]
```diff
@@ -22,7 +22,7 @@ public static void init()
                 {
                     typelist.Add(type);
                 }
-                Run("init", f.Name);
+                Run("init", "", f.Name);
```
Comment (jvyden) [inline jvyterm/PluginHandler.cs]: Try null.

File: jvyterm/PluginHandler.cs [hunk 2]
```diff
@@ -34,15 +34,15 @@ static FileInfo[] getAllDLLs(string dir)
...
+        public static bool Run(string m, string args, string init = null)
         {
             foreach (Type type in typelist)
             {
                 if (m == "init") { logger.log(lang.plugininit2 + init + lang.plugininit3, logger.LogType.Regular); }
                 var c = Activator.CreateInstance(type);
                 try
                 {
-                    type.InvokeMember(m, BindingFlags.InvokeMethod, null, c, null);
+                    type.InvokeMember(m, BindingFlags.InvokeMethod, null, c, new Object[] {args});
```
Comment (jvyden) [inline jvyterm/PluginHandler.cs]: Not sure if that actually passes arguments.

File: jvyterm/shell.cs [hunk 3]
```diff
@@ -23,7 +23,8 @@ static void runCmd(string cmd)
             {
                 foreach (string c in commands)
                 {
-                    if (!PluginHandler.Run(cmd))
+                    String[] arg = cmd.Split(new[] { ' ' }, 2);
+                    if (!PluginHandler.Run(arg[0], arg[1]))
```
Comment (jvyden) [inline jvyterm/shell.cs]: So you're limited to 2 args?

Comment (aepic999) [reply jvyterm/shell.cs]: It gives the command arg to the Run function, and then the rest of the args to the command Run will run

Comment (jvyden) [reply jvyterm/shell.cs]: so arg[0] is the command itself?

File: jvyterm/PluginHandler.cs (hunk 2 above)
Comment (aepic999) [reply jvyterm/PluginHandler.cs]: the microsoft whatever told me it does

File: jvyterm/PluginHandler.cs (hunk 1 above)
Comment (aepic999) [reply jvyterm/PluginHandler.cs]: null isn't a string tho

File: jvyterm/PluginHandler.cs (hunk 2 above)
Comment (jvyden) [reply jvyterm/PluginHandler.cs]: Alright, guess we'll just go with it then

File: jvyterm/PluginHandler.cs (hunk 1 above)
Comment (jvyden) [reply jvyterm/PluginHandler.cs]: It could fuck with the program, though. Its best to just use null or string.null or something

**Comment by jvyden** [comment]
Whoops. made conflicts while commenting.

**Comment by jvyden** [comment]
No updates from OP, closing

**Comment by jvyden** [comment]
Tell me if this needs to be reopened.

**Pull request closed without merging**